docs/source/index.rst

```diff
@@ -52,13 +52,16 @@ that approach may not be as up to date.
 .. toctree::
    :hidden:
    :maxdepth: 1
+   :titlesonly:
 
    user-guide/index
    configuration
    api/index
    commands
    glossary
    dev-guide/index
+   autoapi/conda/index
+   autoapi/conda_env/index
```

Comment:
Let's move this into a new `dev-guide/api` document and link from there, otherwise this'll be confusing for lots of end users

Reply:
https://github.com/conda/conda/pull/12798/commits/66b68e4aa21db8cf727d2b7da9b5155593a54d4d